src/main/java/com/cloudbees/hudson/plugins/folder/AbstractFolder.java

```diff
@@ -324,6 +324,10 @@ public File getRootDirFor(I child) {
         return getRootDirFor(child.getName()); // TODO see comment regarding loadChildren and encoding
     }
 
+    /**
+     * It is unwise to override this, lest links to children from nondefault {@link View}s break.
```

Comment:
Perhaps. I suspect a patch to `View.getDynamic` would suffice. But would need to be tested from various angles, and would require a core update.